MSAL/src/MSALPublicClientApplication.m

```diff
@@ -1434,12 +1435,61 @@ - (void)signoutWithAccount:(nonnull MSALAccount *)account
         block(NO, localError, nil);
         return;
     }
+
+    if (signoutParameters.wipeCacheForAllAccounts)
+    {
+        BOOL result = YES;
+        NSError *localError;
+        
+        result = [self.tokenCache clearCacheForAllAccountsWithContext:nil error:&localError];
+        
+        if (!result)
+        {
+            block(NO, localError, nil);
+            return;
+        }
+        
+#if !TARGET_OS_IPHONE
+        // Clear additional cache locations
+        if (MSALWipeCacheForAllAccountsConfig.additionalPartnerLocations && MSALWipeCacheForAllAccountsConfig.additionalPartnerLocations.count > 0)
+        {
+            NSMutableArray <NSString *> *locationErrors = nil;
+            MSIDMacACLKeychainAccessor *keychainAccessor = [[MSIDMacACLKeychainAccessor alloc] initWithTrustedApplications:nil accessLabel:@"Microsoft Credentials" error:nil];
+            for (NSString* key in MSALWipeCacheForAllAccountsConfig.additionalPartnerLocations)
+            {
+                NSDictionary *cacheLocation = MSALWipeCacheForAllAccountsConfig.additionalPartnerLocations[key];
+                
+                BOOL removeResult = [keychainAccessor removeItemWithAttributes:cacheLocation
+                                                            context:nil
+                                                              error:&localError];
+
+                if (!removeResult)
+                {
+                    result = NO;
+                    if (!locationErrors)
+                    {
+                        locationErrors = [[NSMutableArray alloc] init];
+                    }
+                    [locationErrors addObject:[NSString stringWithFormat:@"'%@'", key]];
+                }
+            }
+            
+            if (!result && locationErrors)
+            {
+                NSError *additionalLocationError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, [NSString stringWithFormat:@"WipeCacheForAllAccounts - error when removing cache for the item(s): %@", [locationErrors componentsJoinedByString:@", "]], nil, nil, localError, nil, nil, YES);
```

Comment:
Can you also return allLocations in userInfo as an array? Might be useful for CP UX.

Reply:
Updated